flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterJobDispatcher.java

```diff
@@ -458,7 +465,14 @@ public JobExecutionResult getResult() throws JobExecutionException, InterruptedE
 				}
 			}
 			else if (result != null) {
-				return result;
+				try {
+					return new SerializedJobExecutionResult(
+						jobId,
+						result.getNetRuntime(),
+						result.getAccumulatorResults()).toJobExecutionResult(ClassLoader.getSystemClassLoader());
```

Comment:
Because the exception is serialized in `OnCompletionActions#jobFailed(JobExecutionResult);`, I have to deserialize it here again. I wonder if this is sane?

Reply:
I think it is ok. Because this is what happens on the client as well.

Code-wise one could also write it
```
return new JobExecutionResult(
	jobId,
	result.getNetRuntime(),
	AccumulatorHelper.deserializeAccumulators(result.getAccumulatorResults(), ClassLoader.getSystemClassLoader()));
```